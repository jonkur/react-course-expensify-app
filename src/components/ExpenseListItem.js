import React from 'react';
import moment from 'moment';
import { Link } from 'react-router-dom';
import numeral from 'numeral';

const ExpenseListItem = ({ description, id, amount, createdAt, note }) => (
    <div>
        <Link to={`/edit/${id}`}>
            <h4>{description}</h4>
        </Link>
        <p>
            Amount: {numeral((amount / 100)).format('$0,0.00')}
             -
            Created at: {moment(createdAt).format('MMMM Do, YYYY')}
             -
            Note: {note}
        </p>
    </div>
);

export default ExpenseListItem;