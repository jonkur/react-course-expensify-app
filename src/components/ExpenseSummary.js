import React from 'react';
import { connect } from 'react-redux';
import selectExpenses from '../selectors/expenses';
import selectExpensesTotal from '../selectors/expenses-total';
import numeral from 'numeral';

export const ExpenseSummary = (props) => (
  <div>
    {props.expenseCount !== 1 ? 
      <p>Viewing {props.expenseCount} expenses totalling {numeral(props.expensesTotal / 100).format('$0,0.00')}</p> : 
      <p>Viewing {props.expenseCount} expense totalling {numeral(props.expensesTotal / 100).format('$0,0.00')}</p>}
  </div>
);

const mapStateToProps = (state) => {
  const visibleExpenses = selectExpenses(state.expenses, state.filters);
  return {
    expenseCount: visibleExpenses.length, 
    expensesTotal: selectExpensesTotal(visibleExpenses)
  };
};

export default connect(mapStateToProps)(ExpenseSummary);