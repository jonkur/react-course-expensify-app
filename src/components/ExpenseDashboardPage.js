import React from 'react';
import ExpenseList from './ExpenseList';
import ExpenseListFilters from './ExpenseListFilters';
import ExpenseSummary from './ExpenseSummary';

const ExpenseDashboardPage = () => (
    <div>
        <ExpenseListFilters />
        <ExpenseList />
        <ExpenseSummary />
    </div>
);

export default ExpenseDashboardPage;