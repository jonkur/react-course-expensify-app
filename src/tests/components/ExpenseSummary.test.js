import React from 'react';
import { shallow } from 'enzyme';

import selectExpensesTotal from '../../selectors/expenses-total';
import { ExpenseSummary } from '../../components/ExpenseSummary'; // named export, pass props
import expenses from '../fixtures/expenses';

test('should render ExpenseSummary correctly', () => {
  const wrapper = shallow(<ExpenseSummary expenseCount={expenses.length} expensesTotal={selectExpensesTotal(expenses)}/>)
  expect(wrapper).toMatchSnapshot();
});

test('should render ExpenseSummary correctly with no data passed in', () => {
  const wrapper = shallow(<ExpenseSummary />);
  expect(wrapper).toMatchSnapshot();
});