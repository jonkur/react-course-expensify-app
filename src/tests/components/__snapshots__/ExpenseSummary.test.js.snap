// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenseSummary correctly 1`] = `
<div>
  <p>
    Viewing 
    3
     expenses totalling 
    $1,141.95
  </p>
</div>
`;

exports[`should render ExpenseSummary correctly with no data passed in 1`] = `
<div>
  <p>
    Viewing 
     expenses totalling 
    $0.00
  </p>
</div>
`;